src/nr5103e_sdk/client.py

```diff
@@ -52,31 +51,38 @@ def __init__(
                 self.username, self.password, self.host = args
         self.verify = verify
         self.timeout = 1
+        self._session: aiohttp.ClientSession | None = None
 
-    def __enter__(self) -> Self:
-        """Do nothing."""
+    async def __aenter__(self) -> Self:
+        """Initialize async session."""
         return self
 
-    def __exit__(
+    async def __aexit__(
         self,
         exc_type: type[BaseException] | None,
         exc_value: BaseException | None,
         traceback: TracebackType | None,
     ) -> bool | None:
-        """Close and delete session from instance cache."""
-        if "session" in self.__dict__:
-            self.session.close()
-            del self.__dict__["session"]
+        """Close session."""
+        if self._session is not None:
+            await self._session.close()
+            self._session = None
         return None
 
-    @cached_property
-    def session(self) -> requests.Session:
-        """Lazy requests session."""
-        session = requests.Session()
-        session.verify = self.verify
-        return session
+    @property
+    def session(self) -> aiohttp.ClientSession:
+        """Lazy aiohttp session."""
+        if self._session is None:
+            verify_ssl = self.verify if self.verify is not None else True
```

Comment:
Change the type of __init__ verify Param to bool instead of `bool | None` to make this check redundant